lib/CMakeLists.txt

```diff
@@ -95,7 +95,14 @@ if(NOT BUILD_SHARED_LIBS)
     set_target_properties(${LIB_NAME} PROPERTIES INTERFACE_COMPILE_DEFINITIONS CURL_STATICLIB)
 endif()
 
-target_link_libraries(${LIB_NAME} ${CURL_LIBS})
+if(BUILD_SHARED_LIBS)
+    target_link_libraries(${LIB_NAME} PRIVATE ${CURL_LIBS})
+else()
+    # It is arguable that the default PUBLIC visibility is correct for
+    # static builds, but it might be a convenient workaround for now,
+    # as correct dependency tracking for static build requires more effort.
+    target_link_libraries(${LIB_NAME} ${CURL_LIBS})
+endif()
```

Comment:
`PRIVATE` should be correct for both shared and static linkage. CMake takes care of this in exported CMake config by adding a `$<LINK_ONLY:target>` generator expression in the `INTERFACE_LINK_LIBRARIES` for private static linkage.

Reply:
indeed. pull request amended to use PRIVATE unconditionally